sc-memory/tests/sc-memory/scs/scs-helper.cpp

```diff
@@ -76,6 +76,42 @@ TEST_F(SCsHelperTest, GenerateBySCs_FileURL)
   EXPECT_EQ(link.GetAsString(), "content: file://test.scs");
 }
 
+TEST_F(SCsHelperTest, GenerateBySCs_ChangeElementType)
+{
+  std::string const data1 = "sc_node_struct -> a;;";
+  std::string const data2 = "sc_node_tuple -> a;;";
+
+  SCsHelper helper(*m_ctx, std::make_shared<TestFileInterface>());
+
+  ScAddr node = m_ctx->CreateNode(ScType::Node);
+  m_ctx->HelperSetSystemIdtf("a", node);
+
+  EXPECT_TRUE(helper.GenerateBySCsText(data1));
+
+  ScType const nodeType = m_ctx->GetElementType(node);
+  EXPECT_EQ(nodeType, ScType::NodeConstStruct);
+
+  EXPECT_FALSE(helper.GenerateBySCsText(data2));
+  EXPECT_EQ(nodeType, ScType::NodeConstStruct);
+}
+
+TEST_F(SCsHelperTest, GenerateBySCs_UsingExistingNonBasicElement)
```

Comment:
Test that test nothing. It fully unusefull in this context.
Where tests that:
- try to change type of `node` to `link`?
- try to change structure type of one element twice in one scs?
- try to setup incorrect type to one element in scs:
  - ```sc_node_struct -> [];;```
  - ```sc_node_material -> [* x -> y;; *];;```
  - ```sc_node_material -> {};;```